common/dev/snoop.c

```diff
@@ -120,10 +110,21 @@ void snoop_read()
 		LOG_ERR("snoop read buffer alloc fail");
 		return;
 	}
-	rc = snoop_aspeed_register_rx_callback(snoop_dev, snoop_rx_callback);
-	if (rc != 0) {
-		printk("Cannot register RX callback\n");
-		return;
+
+	while (1) {
```

Comment:
Earlier this was a callback which was getting invoked, but now u changed this to a blocking read, what's the exit condition here?

Reply:
Hi @jagpalgill 
We noticed that post code may lost when using callback function.
This while loop won't exit, It will keep polling the POST code. Because even if the OS has booted up, the BIOS might still send post codes if some error happens.

Best Regards,
Sean